code/datums/components/self_ignition.dm

```diff
@@ -0,0 +1,57 @@
+/// Component used by plasmeme limbs. Ignites the owner and prevents fire armor from working if they're exposed to oxygen
+/datum/component/self_ignition
```

Comment:
Can be an element I think? No state in this

Reply:
It relies on parent for trait application, so no

Reply:
For just the trait sources? Can't you just use another source, like `[REF(bodypart)]_selfignition"`?